hawkular-wildfly-agent-installer/src/main/java/org/hawkular/wildfly/agent/installer/AgentInstaller.java

```diff
@@ -349,8 +352,21 @@ private static String createSecurityRealm(String keystoreFile, String keystorePa
      */
     private static XmlEdit createStorageAdapter(boolean withHttps, InstallerConfiguration installerConfig) {
         String select = "/server/profile/*[namespace-uri()='urn:org.hawkular.agent:agent:1.0']/";
-        StringBuilder xml = new StringBuilder("<storage-adapter")
-                .append(" type=\"HAWKULAR\"");
+        StringBuilder xml = new StringBuilder("<storage-adapter");
+
+        String tenantId = installerConfig.getTenantId();
+        if (installerConfig.isMetricsOnlyMode()) {
+            xml.append(" type=\"METRICS\"");
+            if (tenantId == null || tenantId.isEmpty()) {
+                throw new IllegalArgumentException("You must specify tenant-id when in metrics-only mode");
+            }
+        } else {
+            xml.append(" type=\"HAWKULAR\"");
+        }
+
+        if (tenantId != null && !tenantId.isEmpty()) {
+            xml.append(" tenantId=\"" + tenantId + "\"");
```

Comment:
Now that I look at the code, the agent only logs a warning but it doesn't halt the agent startup. I don't like that :) There should be an exception thrown here and the agent startup should fail because the agent is starting with a tenant ID that doesn't match the explicit configuration in tenantID. I will fix it so the agent aborts startup after it logs this message.

https://github.com/hawkular/hawkular-agent/blob/master/hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java#L201